IOException: The file 'obj\Debug\net35\annotated\System.Core.xml' already exists.

> error MSB4018: The "AnnotatorBuildTask" task failed unexpectedly.
> System.IO.IOException: The file 'obj\Debug\net35\annotated\System.Core.xml' already exists.

```
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
at System.IO.File.InternalCopy(String sourceFileName, String destFileName, Boolean overwrite, Boolean checkHost)
at System.IO.File.Copy(String sourceFileName, String destFileName)
at TunnelVisionLabs.ReferenceAssemblyAnnotator.AnnotatorBuildTask.Execute()
at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

@sharwell If the file exists, would there ever be a reason that it would need to be updated? Rather than passing `overwrite: true`, maybe the IOException could be swallowed based on the error code (probably ERROR_FILE_EXISTS).

--- a/TunnelVisionLabs.ReferenceAssemblyAnnotator/AnnotatorBuildTask.cs
+++ b/TunnelVisionLabs.ReferenceAssemblyAnnotator/AnnotatorBuildTask.cs
@@ -104,7 +104,14 @@ namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
             string targetDocumentation = Path.ChangeExtension(outputAssembly, ".xml");
             if (File.Exists(sourceDocumentation))
             {
-                File.Copy(sourceDocumentation, targetDocumentation);
+                try
+                {
+                    File.Copy(sourceDocumentation, targetDocumentation);
+                }
+                catch (IOException ex) when ((WindowsErrorCode)ex.HResult == WindowsErrorCode.FileExists)
+                {
+                }
+
                 GeneratedDocumentationFiles = new[] { new TaskItem(targetDocumentation) };
             }
             else

--- a/TunnelVisionLabs.ReferenceAssemblyAnnotator/AnnotatorBuildTask.cs
+++ b/TunnelVisionLabs.ReferenceAssemblyAnnotator/AnnotatorBuildTask.cs
@@ -102,7 +102,7 @@ namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
 
             string sourceDocumentation = Path.ChangeExtension(unannotatedReferenceAssembly, ".xml");
             string targetDocumentation = Path.ChangeExtension(outputAssembly, ".xml");
-            if (File.Exists(sourceDocumentation))
+            if (File.Exists(sourceDocumentation) && !File.Exists(targetDocumentation))
             {
                 try
                 {

--- a/TunnelVisionLabs.ReferenceAssemblyAnnotator/WindowsErrorCode.cs
+++ b/TunnelVisionLabs.ReferenceAssemblyAnnotator/WindowsErrorCode.cs
@@ -1,0 +1,13 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
+{
+    internal enum WindowsErrorCode : ushort
+    {
+        /// <summary>
+        /// https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-#error_file_exists
+        /// </summary>
+        FileExists = 80,
+    }
+}

